sdk/storage/azure-storage-blob-batch/src/main/java/com/azure/storage/blob/batch/BlobBatchHelper.java

```diff
@@ -47,16 +60,33 @@
         String contentType = batchResponse.getDeserializedHeaders().getContentType();
 
         // Split on the boundary [ "multipart/mixed; boundary", "batchresponse_66925647-d0cb-4109-b6d3-28efe3e1e5ed"]
-        String boundary = contentType.split("=", 2)[1];
+        String[] boundaryPieces = contentType.split("=", 2);
+        if (boundaryPieces.length == 1) {
+            return Mono.error(logger
+                .logExceptionAsError(new IllegalStateException("Response doesn't contain a boundary.")));
+        }
+
+        String boundary = boundaryPieces[1];
 
         return FluxUtil.collectBytesInByteBufferStream(batchResponse.getValue())
-            .flatMap(byteArrayBody -> {
+            .flatMap(byteArrayBody -> Mono.fromRunnable(() -> {
                 String body = new String(byteArrayBody, StandardCharsets.UTF_8);
+                List<BlobStorageException> exceptions = new ArrayList<>();
+
+                String[] subResponses = body.split("--" + boundary);
+                if (subResponses.length == 3 && batch.getOperationCount() != 1) {
+                    String[] exceptionSections = subResponses[1].split(HTTP_NEWLINE + HTTP_NEWLINE);
```

Comment:
Some of these I'm confused about, and may require some docs like.. why do we split on two new lines?

Reply:
I'll comment this, HTTP responses split their metadata and body with `\r\n\r\n`.

Reply:
I will add an issue to change this next week as it won't touch API

Reply:
https://github.com/Azure/azure-sdk-for-java/issues/5942